### PR TITLE
Add support for discovery

### PR DESCRIPTION
**Pull request by DRSchlaubi**
Discord PR: discord/discord-api-docs#1762

### CODE REVIEW

**Comment by HopeBaron** [comment]
where is the PR that caused this change?

**Comment by DRSchlaubi** [comment]
Whoops totally forgot to link it

File: common/src/main/kotlin/entity/DiscordDiscoveryMetadata.kt [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+import dev.kord.common.entity.optional.Optional
+import dev.kord.common.entity.optional.OptionalBoolean
+import dev.kord.common.entity.optional.OptionalSnowflake
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+@Serializable
+data class DiscordDiscoveryMetadata(
+    @SerialName("guild_id")
```
Comment (HopeBaron) [inline common/src/main/kotlin/entity/DiscordDiscoveryMetadata.kt:11]: Missing primary_category_id field

File: common/src/main/kotlin/entity/DiscordDiscoveryMetadata.kt [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+) {
+    @Serializable
+    data class Name(
+        val default: String,
+        val localizations: Optional<Map<String, String>>
+    )
+}
+
+@Serializable
+data class DiscoveryTermValidationResponse(val valid: Boolean)
```
Comment (HopeBaron) [inline common/src/main/kotlin/entity/DiscordDiscoveryMetadata.kt:39]: Move this to rest module under json then response.

Edit: As I'm reviewing I realized that these are requests

File: common/src/main/kotlin/entity/DiscordDiscoveryMetadata.kt [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+@Serializable
+data class PartialDiscordDiscoveryMetadata(
```
Comment (HopeBaron) [inline common/src/main/kotlin/entity/DiscordDiscoveryMetadata.kt:42]: Same as above

File: common/src/main/kotlin/entity/DiscordDiscoveryMetadata.kt [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    @SerialName("primary_category_id")
+    val primaryCategoryId: OptionalSnowflake = OptionalSnowflake.Missing,
+    val keywords: Optional<List<String>> = Optional.Missing(),
+    @SerialName("emoji_discoverability_enabled")
+    val emojiDiscoverabilityEnabled: OptionalBoolean = OptionalBoolean.Missing
+)
+
+@Serializable
+data class AddGuildDiscoverySubCategoryResponse(
```
Comment (HopeBaron) [inline common/src/main/kotlin/entity/DiscordDiscoveryMetadata.kt:51]: Same as above

**Review by HopeBaron — COMMENTED** [review]
This is great, all that's left is the core entities.
I added some suggestions

**Comment by DRSchlaubi** [comment]
> This is great, all that's left is the core entities.
> I added some suggestions

Looks like the 2nd try to push the core enities dind't work too

**Comment by HopeBaron** [comment]
What do you mean

**Comment by DRSchlaubi** [comment]
> What do you mean

there already are core entities 
git just didn't want me to push them

**Comment by DRSchlaubi** [comment]
How is it that the Discord Docs pr for this is still not merged

**Pull request closed without merging**